scarb/src/ops/resolve.rs

```diff
@@ -219,7 +221,25 @@ fn generate_cairo_compilation_units(
                         package
                     };
 
-                    CompilationUnitComponent { package, target }
+                    let cfg_set = {
+                        if package.id.name.to_smol_str() == main_component_name {
+                            None
+                        } else {
```

Comment:
I am not sure if we are safe to remove `cfg(test)` from corelib for the time being. Have you verified that? Will follow up offline.

Reply:
I've done some tests that prompted me to remove this check but I'm also not so sure about it. Basically tests that were previously failing because of that were passing again.